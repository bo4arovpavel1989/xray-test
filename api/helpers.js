const jwt = require('jsonwebtoken');
const secret = require('./credentials');
const sizeOfCallback = require('image-size');
const db = require('./dbqueries');

/**
 * Return JSON web token, based on response object
 * @param {object} res The response object.
 * @returns {string} The JSON web token generated on response data
 */
const setToken = res => jwt.sign({ login: res.login, pass: res.passwd, email: res.email, date: new Date() }, secret.secret)

module.exports.setToken = setToken

const sizeOf = function (file) {
  return new Promise((resolve, reject) => {
      sizeOfCallback(file, (err, dimensions) => {
        if (!err) resolve(dimensions)
        else reject(err)
      })
  })
};

module.exports.sizeOf = sizeOf;

/**
 * Function generates array of db update queries for settings
 * to use it in Promise.all
 * @params {Array} settings - got from client
 * @returns {Array} - db update queries
 */
const getSettingsQueryArray = function (settings) {
  const settingsQueryArray = [];

  settings.forEach(tune => {
    const { name } = tune;

    settingsQueryArray.push(db.update('Settings', { name }, tune))
  });

  return settingsQueryArray;
}

module.exports.getSettingsQueryArray = getSettingsQueryArray;
