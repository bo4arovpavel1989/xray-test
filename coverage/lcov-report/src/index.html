<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> src
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.55% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>13/44</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.71% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>13/41</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="actions.js"><a href="actions.js.html">actions.js</a></td>
	<td data-value="20.83" class="pic low"><div class="chart"><div class="cover-fill" style="width: 20%;"></div><div class="cover-empty" style="width:80%;"></div></div></td>
	<td data-value="20.83" class="pct low">20.83%</td>
	<td data-value="24" class="abs low">5/24</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="19" class="abs low">0/19</td>
	<td data-value="20.83" class="pct low">20.83%</td>
	<td data-value="24" class="abs low">5/24</td>
	</tr>

<tr>
	<td class="file low" data-value="config.js"><a href="config.js.html">config.js</a></td>
	<td data-value="40" class="pic low"><div class="chart"><div class="cover-fill" style="width: 40%;"></div><div class="cover-empty" style="width:60%;"></div></div></td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="10" class="abs low">4/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	</tr>

<tr>
	<td class="file low" data-value="helpers.js"><a href="helpers.js.html">helpers.js</a></td>
	<td data-value="40" class="pic low"><div class="chart"><div class="cover-fill" style="width: 40%;"></div><div class="cover-empty" style="width:60%;"></div></div></td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="10" class="abs low">4/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Apr 17 2019 19:14:01 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
