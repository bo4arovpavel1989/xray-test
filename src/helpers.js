/**
 * Fetch method helper makes json transformation
 * @param {object} response - response object got from server
 * @returns {Promise} - boolean represents if json operation success
 */
export const handleResponse = response => response.json().then(json => response.ok ? json : Promise.reject(json))
