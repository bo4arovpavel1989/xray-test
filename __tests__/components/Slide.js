import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import { Slide } from '../../src/components/Test/Slide.js';
import Enzyme, { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { comments } from '../../src/helpers';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Slide component', () => {
  const props = {
    settings: {},
    question: {},
    sendResult: jest.fn(),
    nextQuestion: jest.fn(),
    drawer: {
      reset: jest.fn(),
      start: jest.fn(),
      clearZones: jest.fn(),
      getZones: jest.fn(() => [[0, 0, 100, 100]]),
      drawOldZones: jest.fn()
    },
    prepareCanvas:  jest.fn()
  };

  const initialState = {
    answered: false,
    canvasBackground: '#canvasBackground',
    canvasDraw: '#canvasDrawArea',
    comment: comments.right,
    photoShowed: false,
    question: {},
    result: 0,
    settings: {},
    slideShowed: true,
    warningShowed: false
  };

  const component = mount(
    <MemoryRouter initialEntries={[ { key: 'testKey' } ]}>
      <Slide { ...props }/>
    </MemoryRouter>
  );

  describe('should render correct', () => {
      afterAll(() => {
        props.prepareCanvas.mockClear();
      });

      it('match snapshot', () => {
        expect(shallowToJson(component)).toMatchSnapshot()
      })

      it('should have initial state', () => {
        expect(component.find(Slide).instance().state).toEqual(initialState)
      })

      it('should call prepareCanvas', () => {
        expect(props.prepareCanvas).toHaveBeenCalledTimes(1);
      })
  })

  describe('should handle question warning timer', () => {
      jest.useFakeTimers();

      beforeAll(() => {
        component.find(Slide).instance().setState({
          settings: {
            timerWarning: 10,
            time: 15,
            redError: 8
          },
          question: {
            dangerZones: [0, 0, 100, 100],
            isDanger: '1'
          }
        });
        component.find(Slide).instance().setQuestionTimers();

        jest.runAllTimers();
      });
      afterAll(() => {
        component.find(Slide).instance().setState(initialState);
        props.drawer.drawOldZones.mockClear();
      });


      it('match snapshot', () => {
        expect(shallowToJson(component)).toMatchSnapshot()
      })

      it('should change state - warningShowed', () => {
        expect(component.find(Slide).instance().state.warningShowed).toEqual(true)
      })
      it('should change state - comment', () => {
        expect(component.find(Slide).instance().state.comment).toEqual(comments.time)
      })
      it('should change state - answered', () => {
        expect(component.find(Slide).instance().state.answered).toEqual(true)
      })
      it('should change state - result', () => {
        expect(component.find(Slide).instance().state.result).toEqual(8)
      })

      it('should call drawZones', () => {
        expect(props.drawer.drawOldZones).toHaveBeenCalledTimes(1)
      })
  })

})
