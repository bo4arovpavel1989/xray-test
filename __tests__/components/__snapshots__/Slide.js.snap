// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide component should handle canvas click - when danger - right click match snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Slide
      drawer={
        Object {
          "clearZones": [MockFunction],
          "drawOldZones": [MockFunction] {
            "calls": Array [
              Array [
                Array [
                  Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                ],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "getZones": [MockFunction],
          "reset": [MockFunction],
          "start": [MockFunction],
        }
      }
      nextQuestion={[MockFunction]}
      prepareCanvas={[MockFunction]}
      question={Object {}}
      sendResult={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "click": undefined,
                "comment": "Время истекло!",
                "question": Object {
                  "dangerZones": Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
            Array [
              Object {
                "click": null,
                "comment": "Ошибка! Опасный предмет есть!",
                "question": Object {
                  "dangerZones": Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
            Array [
              Object {
                "click": null,
                "comment": "Поздравляем! Правильный ответ.",
                "question": Object {
                  "dangerZones": Array [],
                  "isDanger": "0",
                },
                "result": 0,
              },
            ],
            Array [
              Object {
                "click": Object {
                  "x": undefined,
                  "y": undefined,
                },
                "comment": "Ошибка! Опасных предметов нет!",
                "question": Object {
                  "dangerZones": Array [],
                  "isDanger": "0",
                },
                "result": 4,
              },
            ],
            Array [
              Object {
                "click": Object {
                  "x": 101,
                  "y": 101,
                },
                "comment": "Ошибка! Опасного предмета нет в обозначенной Вами области!",
                "question": Object {
                  "dangerZones": Array [
                    Array [
                      0,
                      0,
                      100,
                      100,
                    ],
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
            Array [
              Object {
                "click": Object {
                  "x": 50,
                  "y": 50,
                },
                "comment": "Поздравляем! Правильный ответ.",
                "question": Object {
                  "dangerZones": Array [
                    Array [
                      0,
                      0,
                      100,
                      100,
                    ],
                  ],
                  "isDanger": "1",
                },
                "result": 0,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      settings={Object {}}
    >
      <div>
        <div>
          Вопрос 
        </div>
        <div
          className="canvasArea"
        >
          <div
            className="actionComment_container speech"
          >
            <div
              className="green"
              id="actionComment"
            >
              Поздравляем! Правильный ответ.
            </div>
          </div>
          <div
            className="markIcon_cotainer"
          >
            <span
              className="greenMark"
              id="markIcon"
            />
          </div>
          <div
            className="clearButton_container"
          >
            <button
              disabled={true}
              id="clearButton"
              onClick={[Function]}
            >
              ☑
            </button>
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              className="animatedShow"
              id="canvasBackground"
            />
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              id="canvasDrawArea"
              onClick={[Function]}
            />
          </div>
          <div
            className="timerWarning_container"
          />
          <div
            className="forwardButton_container"
          >
            <button
              disabled={true}
              id="forwardButton"
              onClick={[Function]}
            >
              ►
            </button>
          </div>
        </div>
        <img
          alt="Опасный предмет"
          className="dangerPicture"
        />
      </div>
    </Slide>
  </Router>
</MemoryRouter>
`;

exports[`Slide component should handle canvas click - when danger - wrong click match snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Slide
      drawer={
        Object {
          "clearZones": [MockFunction],
          "drawOldZones": [MockFunction] {
            "calls": Array [
              Array [
                Array [
                  Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                ],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "getZones": [MockFunction],
          "reset": [MockFunction],
          "start": [MockFunction],
        }
      }
      nextQuestion={[MockFunction]}
      prepareCanvas={[MockFunction]}
      question={Object {}}
      sendResult={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "click": undefined,
                "comment": "Время истекло!",
                "question": Object {
                  "dangerZones": Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
            Array [
              Object {
                "click": null,
                "comment": "Ошибка! Опасный предмет есть!",
                "question": Object {
                  "dangerZones": Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
            Array [
              Object {
                "click": null,
                "comment": "Поздравляем! Правильный ответ.",
                "question": Object {
                  "dangerZones": Array [],
                  "isDanger": "0",
                },
                "result": 0,
              },
            ],
            Array [
              Object {
                "click": Object {
                  "x": undefined,
                  "y": undefined,
                },
                "comment": "Ошибка! Опасных предметов нет!",
                "question": Object {
                  "dangerZones": Array [],
                  "isDanger": "0",
                },
                "result": 4,
              },
            ],
            Array [
              Object {
                "click": Object {
                  "x": 101,
                  "y": 101,
                },
                "comment": "Ошибка! Опасного предмета нет в обозначенной Вами области!",
                "question": Object {
                  "dangerZones": Array [
                    Array [
                      0,
                      0,
                      100,
                      100,
                    ],
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      settings={Object {}}
    >
      <div>
        <div>
          Вопрос 
        </div>
        <div
          className="canvasArea"
        >
          <div
            className="actionComment_container speech"
          >
            <div
              className="red"
              id="actionComment"
            >
              Ошибка! Опасного предмета нет в обозначенной Вами области!
            </div>
          </div>
          <div
            className="markIcon_cotainer"
          >
            <span
              className="redError"
              id="markIcon"
            />
          </div>
          <div
            className="clearButton_container"
          >
            <button
              disabled={true}
              id="clearButton"
              onClick={[Function]}
            >
              ☑
            </button>
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              className="animatedShow"
              id="canvasBackground"
            />
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              id="canvasDrawArea"
              onClick={[Function]}
            />
          </div>
          <div
            className="timerWarning_container"
          />
          <div
            className="forwardButton_container"
          >
            <button
              disabled={false}
              id="forwardButton"
              onClick={[Function]}
            >
              ►
            </button>
          </div>
        </div>
        <img
          alt="Опасный предмет"
          className="hidden"
        />
      </div>
    </Slide>
  </Router>
</MemoryRouter>
`;

exports[`Slide component should handle canvas click - when safe match snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Slide
      drawer={
        Object {
          "clearZones": [MockFunction],
          "drawOldZones": [MockFunction],
          "getZones": [MockFunction],
          "reset": [MockFunction],
          "start": [MockFunction],
        }
      }
      nextQuestion={[MockFunction]}
      prepareCanvas={[MockFunction]}
      question={Object {}}
      sendResult={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "click": undefined,
                "comment": "Время истекло!",
                "question": Object {
                  "dangerZones": Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
            Array [
              Object {
                "click": null,
                "comment": "Ошибка! Опасный предмет есть!",
                "question": Object {
                  "dangerZones": Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
            Array [
              Object {
                "click": null,
                "comment": "Поздравляем! Правильный ответ.",
                "question": Object {
                  "dangerZones": Array [],
                  "isDanger": "0",
                },
                "result": 0,
              },
            ],
            Array [
              Object {
                "click": Object {
                  "x": undefined,
                  "y": undefined,
                },
                "comment": "Ошибка! Опасных предметов нет!",
                "question": Object {
                  "dangerZones": Array [],
                  "isDanger": "0",
                },
                "result": 4,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      settings={Object {}}
    >
      <div>
        <div>
          Вопрос 
        </div>
        <div
          className="canvasArea"
        >
          <div
            className="actionComment_container speech"
          >
            <div
              className="yellow"
              id="actionComment"
            >
              Ошибка! Опасных предметов нет!
            </div>
          </div>
          <div
            className="markIcon_cotainer"
          >
            <span
              className="yellowError"
              id="markIcon"
            />
          </div>
          <div
            className="clearButton_container"
          >
            <button
              disabled={true}
              id="clearButton"
              onClick={[Function]}
            >
              ☑
            </button>
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              className="animatedShow"
              id="canvasBackground"
            />
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              id="canvasDrawArea"
              onClick={[Function]}
            />
          </div>
          <div
            className="timerWarning_container"
          />
          <div
            className="forwardButton_container"
          >
            <button
              disabled={false}
              id="forwardButton"
              onClick={[Function]}
            >
              ►
            </button>
          </div>
        </div>
        <img
          alt="Опасный предмет"
          className="hidden"
        />
      </div>
    </Slide>
  </Router>
</MemoryRouter>
`;

exports[`Slide component should handle clear-button click - when danger match snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Slide
      drawer={
        Object {
          "clearZones": [MockFunction],
          "drawOldZones": [MockFunction] {
            "calls": Array [
              Array [
                Array [
                  0,
                  0,
                  100,
                  100,
                ],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "getZones": [MockFunction],
          "reset": [MockFunction],
          "start": [MockFunction],
        }
      }
      nextQuestion={[MockFunction]}
      prepareCanvas={[MockFunction]}
      question={Object {}}
      sendResult={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "click": undefined,
                "comment": "Время истекло!",
                "question": Object {
                  "dangerZones": Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
            Array [
              Object {
                "click": null,
                "comment": "Ошибка! Опасный предмет есть!",
                "question": Object {
                  "dangerZones": Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      settings={Object {}}
    >
      <div>
        <div>
          Вопрос 
        </div>
        <div
          className="canvasArea"
        >
          <div
            className="actionComment_container speech"
          >
            <div
              className="red"
              id="actionComment"
            >
              Ошибка! Опасный предмет есть!
            </div>
          </div>
          <div
            className="markIcon_cotainer"
          >
            <span
              className="redError"
              id="markIcon"
            />
          </div>
          <div
            className="clearButton_container"
          >
            <button
              disabled={true}
              id="clearButton"
              onClick={[Function]}
            >
              ☑
            </button>
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              className="animatedShow"
              id="canvasBackground"
            />
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              id="canvasDrawArea"
              onClick={[Function]}
            />
          </div>
          <div
            className="timerWarning_container"
          />
          <div
            className="forwardButton_container"
          >
            <button
              disabled={false}
              id="forwardButton"
              onClick={[Function]}
            >
              ►
            </button>
          </div>
        </div>
        <img
          alt="Опасный предмет"
          className="hidden"
        />
      </div>
    </Slide>
  </Router>
</MemoryRouter>
`;

exports[`Slide component should handle clear-button click - when safe match snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Slide
      drawer={
        Object {
          "clearZones": [MockFunction],
          "drawOldZones": [MockFunction],
          "getZones": [MockFunction],
          "reset": [MockFunction],
          "start": [MockFunction],
        }
      }
      nextQuestion={[MockFunction]}
      prepareCanvas={[MockFunction]}
      question={Object {}}
      sendResult={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "click": undefined,
                "comment": "Время истекло!",
                "question": Object {
                  "dangerZones": Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
            Array [
              Object {
                "click": null,
                "comment": "Ошибка! Опасный предмет есть!",
                "question": Object {
                  "dangerZones": Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
            Array [
              Object {
                "click": null,
                "comment": "Поздравляем! Правильный ответ.",
                "question": Object {
                  "dangerZones": Array [],
                  "isDanger": "0",
                },
                "result": 0,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      settings={Object {}}
    >
      <div>
        <div>
          Вопрос 
        </div>
        <div
          className="canvasArea"
        >
          <div
            className="actionComment_container speech"
          >
            <div
              className="green"
              id="actionComment"
            >
              Поздравляем! Правильный ответ.
            </div>
          </div>
          <div
            className="markIcon_cotainer"
          >
            <span
              className="greenMark"
              id="markIcon"
            />
          </div>
          <div
            className="clearButton_container"
          >
            <button
              disabled={true}
              id="clearButton"
              onClick={[Function]}
            >
              ☑
            </button>
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              className="animatedShow"
              id="canvasBackground"
            />
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              id="canvasDrawArea"
              onClick={[Function]}
            />
          </div>
          <div
            className="timerWarning_container"
          />
          <div
            className="forwardButton_container"
          >
            <button
              disabled={false}
              id="forwardButton"
              onClick={[Function]}
            >
              ►
            </button>
          </div>
        </div>
        <img
          alt="Опасный предмет"
          className="hidden"
        />
      </div>
    </Slide>
  </Router>
</MemoryRouter>
`;

exports[`Slide component should handle question warning timer match snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Slide
      drawer={
        Object {
          "clearZones": [MockFunction],
          "drawOldZones": [MockFunction] {
            "calls": Array [
              Array [
                Array [
                  0,
                  0,
                  100,
                  100,
                ],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "getZones": [MockFunction],
          "reset": [MockFunction],
          "start": [MockFunction],
        }
      }
      nextQuestion={[MockFunction]}
      prepareCanvas={[MockFunction]}
      question={Object {}}
      sendResult={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "click": undefined,
                "comment": "Время истекло!",
                "question": Object {
                  "dangerZones": Array [
                    0,
                    0,
                    100,
                    100,
                  ],
                  "isDanger": "1",
                },
                "result": 8,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      settings={Object {}}
    >
      <div>
        <div>
          Вопрос 
        </div>
        <div
          className="canvasArea"
        >
          <div
            className="actionComment_container speech"
          />
          <div
            className="markIcon_cotainer"
          />
          <div
            className="clearButton_container"
          >
            <button
              disabled={true}
              id="clearButton"
              onClick={[Function]}
            >
              ☑
            </button>
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              className="animatedShow"
              id="canvasBackground"
            />
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              id="canvasDrawArea"
              onClick={[Function]}
            />
          </div>
          <div
            className="timerWarning_container"
          />
          <div
            className="forwardButton_container"
          >
            <button
              disabled={true}
              id="forwardButton"
              onClick={[Function]}
            >
              ►
            </button>
          </div>
        </div>
        <img
          alt="Опасный предмет"
          className="hidden"
        />
      </div>
    </Slide>
  </Router>
</MemoryRouter>
`;

exports[`Slide component should render correct match snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Slide
      drawer={
        Object {
          "clearZones": [MockFunction],
          "drawOldZones": [MockFunction],
          "getZones": [MockFunction],
          "reset": [MockFunction],
          "start": [MockFunction],
        }
      }
      nextQuestion={[MockFunction]}
      prepareCanvas={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      question={Object {}}
      sendResult={[MockFunction]}
      settings={Object {}}
    >
      <div>
        <div>
          Вопрос 
        </div>
        <div
          className="canvasArea"
        >
          <div
            className="actionComment_container speech"
          />
          <div
            className="markIcon_cotainer"
          />
          <div
            className="clearButton_container"
          >
            <button
              disabled={true}
              id="clearButton"
              onClick={[Function]}
            >
              ☑
            </button>
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              className="animatedShow"
              id="canvasBackground"
            />
          </div>
          <div
            className="canvas_container"
          >
            <canvas
              id="canvasDrawArea"
              onClick={[Function]}
            />
          </div>
          <div
            className="timerWarning_container"
          />
          <div
            className="forwardButton_container"
          >
            <button
              disabled={true}
              id="forwardButton"
              onClick={[Function]}
            >
              ►
            </button>
          </div>
        </div>
        <img
          alt="Опасный предмет"
          className="hidden"
        />
      </div>
    </Slide>
  </Router>
</MemoryRouter>
`;
